Lib/test/test_dataclasses/__init__.py

```diff
@@ -3664,6 +3664,34 @@ class A(WithDictSlot): ...
         self.assertEqual(A().__dict__, {})
         A()
 
+    def test_slots_with_wrong_init_subclass(self):
```

Comment:
The value of adding tests for buggy behavior is not clear to me, but if we will add the tests, I'd rather have a comment making it very clear that this tested behavior is a bug that should ideally be fixed, not desired behavior.